youtube_dl/extractor/npo.py

```diff
@@ -170,6 +170,9 @@ def _get_info(self, video_id):
             transform_source=strip_jsonp,
         )
 
+        if 'error' in metadata and metadata['code'] == 440:
+            raise ExtractorError('Video gone')
```

Comment:
There is a clear human readable error provided that must be used. Also must be `expected`.